File: demo-task-executor-java/src/test/java/demo/executor/AbstractConcurrencyTest.java
package demo.executor;

import static org.testng.Assert.assertTrue;

/**
 * Abstract unit test for testing concurrency.
 */
public class AbstractConcurrencyTest {

    private static final int RELATIVE_ERROR_PERCENT = 5;

    protected void assertAlmostEquals(long actual, long expected) {
        assertTrue(Math.abs(100 * (actual - expected) / expected) < RELATIVE_ERROR_PERCENT);
    }
}

File: demo-task-executor-kotlin/src/test/kotlin/demo/executor/AbstractConcurrencyTest.kt
package demo.executor

import org.testng.Assert.assertTrue

/**
 * Abstract unit test for testing concurrency.
 */
open class AbstractConcurrencyTest {

    private val RELATIVE_ERROR_PERCENT = 5

    protected fun assertAlmostEquals(actual: Long, expected: Long) {
        assertTrue(Math.abs(100 * (actual - expected) / expected) < RELATIVE_ERROR_PERCENT)
    }
}
